### PR TITLE
drivers: clock_control: gd32: initial support

### LINKED ISSUE #49074: GD32: Use clocks instead rcu-periph-clock property
**Is your enhancement proposal related to a problem? Please describe.**
The `rcu-periph-clock` property is not well align with devicetree and should be replaced. The correct property for that use is `clocks`.
This is a step closer to have a full clock capability on the GD32 SoCs family. 

**Describe the solution you'd like**
Replace `rcu-periph-clock` property by correct `clocks` property. The change itself not require a implementation of clock driver to allow the fix.

**Additional context**
The solution can mimics the following [implementation](https://elixir.bootlin.com/linux/v5.19/source/Documentation/devicetree/bindings/spi/atmel,quadspi.yaml#L85) from Linux. A reference from `&rcu` node + `RCU_TYPE_PERIPHERAL` + clock identificator. The change to correct the property not seems to represent a high effort in the first look.

CC: @gmarull @cameled @soburi

### PR DESCRIPTION
**Pull request by gmarull**
This patch adds a clock control driver for GD32 platforms. It is
important to note that the driver can only handle peripheral
clocks, but not "system clocks" (e.g. PLL settings, SYS_CK, etc.).  On
some similar platforms (STM32) this task is embedded in the same clock
driver, performed at init time but with no options to do any
manipulation at runtime via the API calls. The clock control API as-is
is really orthogonal to "system clocks", and it is arguably a bad idea
to embed system clock init code in a clock control driver. It can be
done at SoC level still using Devicetree as a source of hardware
description/initial configuration. This situation could change in the future
if we ever have a proper clock control API.

Quickly tested on gd32f450v_start and gd32e507v_start boards (serial/gpio/pinctrl/pwm)

Fixes #49074

Signed-off-by: Gerard Marull-Paretas <gerard@teslabs.com>

### CODE REVIEW

File: dts/bindings/clock/gd,gd32-cctl.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Copyright (c) 2022, Teslabs Engineering S.L.
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  Gigadevice Reset and Clock Unit (RCU) if a multi-function peripheral in
+  charge of reset control (RCTL) and clock control (CCTL) for all SoC
+  peripherals. This binding represents the clock controller (CCTL).
+
+  To specify the reset line in a peripheral, the standard resets property needs
```
Comment (cameled) [inline dts/bindings/clock/gd,gd32-cctl.yaml:9]: update `reset ` to `clock`.

**Review by cameled — COMMENTED** [review]
spi_flash sample run success  on my gd32vf103v_eval and gd32f450i_eval.

**Review by cameled — DISMISSED** [review]
LGTM, really nice work.

**Review by nandojve — DISMISSED** [review]
Nice work!

**Comment by gmarull** [comment]
rebased to fix conflicts, @nandojve @cameled please re-approve

**Review by nandojve — APPROVED** [review]

**Review by cameled — APPROVED** [review]

**Pull request merged**